collection-scripts/gather_virtualmachines

```diff
@@ -1,33 +1,27 @@
-#!/bin/bash -x
+#!/bin/bash
 
-BASE_COLLECTION_PATH="${BASE_COLLECTION_PATH:-/must-gather}"
+export BASE_COLLECTION_PATH="${BASE_COLLECTION_PATH:-/must-gather}"
 PROS=${PROS:-5}
 
 function collect_vm() {
-  vm=$1
-  ns=$2
-  ns_path=$3
-
-  if oc get -n ${ns} vm ${vm} -o jsonpath='{ .metadata.annotations }' | grep 'vm.kubevirt.io/template'; then
-    vm_path="$ns_path/template-based"
+  item=$(echo $1 | base64 -d)
+  ns=$(echo ${item} | jq -r '.metadata.namespace')
+  ns_path="${BASE_COLLECTION_PATH}/namespaces/${ns}/kubevirt.io/virtualmachines"
+  name=$(echo ${item} | jq -r '.metadata.name')
+  echo "Reading VM ${ns}/${name}"
+  if echo ${item} | jq -e '.metadata.annotation["vm.kubevirt.io/template"] == null'; then
+    dir="${ns_path}/custom/"
   else
-    vm_path="$ns_path/custom"
+    dir="${ns_path}/template-based/"
   fi
-
-  mkdir -p "${vm_path}"
-
-  /usr/bin/oc get virtualmachine "$vm" -o yaml -n "$ns" > "$vm_path/$vm.yaml"
+  mkdir -p ${dir}
+  echo ${item} | yq eval -P - > "${dir}/${name}.yaml"
 }
 
 export -f collect_vm
 
-namespaces=($(/usr/bin/oc get virtualmachines --all-namespaces --no-headers | awk '{print $1}' | uniq))
-
-for ns in "${namespaces[@]}"; do
-  ns_path="${BASE_COLLECTION_PATH}/namespaces/${ns}/kubevirt.io/virtualmachines"
+ALL_VMS_JSON=$(/usr/bin/oc get virtualmachine -A -o json)
 
-  vms=$(/usr/bin/oc get virtualmachine -n "${ns}" --no-headers -o custom-columns=NAME:.metadata.name)
-  echo ${vms[@]} | tr ' ' '\n' | xargs -t -P ${PROS} --max-args=1 -I{} sh -c 'collect_vm $1 $2 $3' -- {} "${ns}" "${ns_path}"
-done
+echo $ALL_VMS_JSON | jq -r '.items[] | @base64' | xargs -P ${PROS} --max-args=1 -I{} sh -c 'collect_vm  $1' -- {}
```

Comment:
I suspect quoting would be safer (unless you WANT to break on spaces, quoting variables should be the default in bash)
```suggestion
echo "$ALL_VMS_JSON" | jq -r '.items[] | @base64' | xargs -P ${PROS} --max-args=1 -I{} sh -c 'collect_vm  $1' -- {}
```

but frankly, since this string can be quite long and consume much memory, why do you need the temporary variable at all? why not pipe it from `oc get vms` ? (I'm not sure about this myself)